### PR TITLE
THEEDGE-2960 Return the number of updates available on ImageInfo

### PR DESCRIPTION
**Pull request by de1987**
# Description

Added `Count` on `ImageInfo` to return the number of updates available as requested on [THEEDGE-2923](https://issues.redhat.com/browse/THEEDGE-2923)
Fixes # [(THEEDGE-2960)](https://issues.redhat.com/browse/THEEDGE-2960)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [x] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/models/images.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ type PackageDiff struct {
 type ImageInfo struct {
 	Image            Image                   `json:"Image"`
 	UpdatesAvailable *[]ImageUpdateAvailable `json:"UpdatesAvailable,omitempty"`
+	Count            int64                   `json:"Count"`
```
Comment (omaciel) [inline pkg/models/images.go:71]: Would it make sense to add a simple test for the model itself to ensure that we have a `Count` attribute?

**Review by omaciel — APPROVED** [review]
/lgtm

Please consider my comment about adding a simple test

File: pkg/models/images.go (hunk 1 above)
Comment (de1987) [reply pkg/models/images.go:71]: @omaciel done partially. I'm gonna improve the test's asserts

Comment (de1987) [reply pkg/models/images.go:71]: @omaciel done

File: pkg/routes/devices.go [hunk 2]
```diff
@@ -159,7 +159,7 @@ func GetUpdateAvailableForDevice(w http.ResponseWriter, r *http.Request) {
 		latest = true
 	}
 	pagination := common.GetPagination(r)
-	result, err := contextServices.DeviceService.GetUpdateAvailableForDeviceByUUID(dc.DeviceUUID, latest, pagination.Limit, pagination.Offset)
+	result, _, err := contextServices.DeviceService.GetUpdateAvailableForDeviceByUUID(dc.DeviceUUID, latest, pagination.Limit, pagination.Offset)
```
Comment (mgold1234) [inline pkg/routes/devices.go:162]: why is this _ necessary here?

**Review by mgold1234 — COMMENTED** [review]
look good, left one comment.
I dont see tests for count field that you add in models

File: pkg/routes/devices.go (hunk 2 above)
Comment (de1987) [reply pkg/routes/devices.go:162]: @mgold1234 the function is returning 3 values, the second is the count for the update available but this route specifically does not use it.
To add to this route would require changing the API contract.

File: pkg/routes/devices_test.go [hunk 3]
```diff
@@ -209,6 +213,192 @@ var _ = Describe("Devices View Router", func() {
...
+				Expect(deviceImageUpdate.Image.Distribution).To(Equal(imageUpdate.Distribution))
+				Expect(deviceImageUpdate.Image.Commit.OSTreeCommit).To(Equal(imageUpdate.Commit.OSTreeCommit))
+
+				Expect(deviceResponse.OrgID).To(Equal(device.OrgID))
+				Expect(deviceResponse.UUID).To(Equal(device.UUID))
+				Expect(deviceResponse.Name).To(Equal(device.Name))
+				Expect(deviceResponse.RHCClientID).To(Equal(device.RHCClientID))
+				Expect(deviceResponse.ImageID).To(Equal(image.ID))
+
+				Expect(deviceDetails.Image.Count).To(Equal(int64(1)))
```
Comment (de1987) [inline pkg/routes/devices_test.go:394]: @mgold1234 I think this is what you are looking for

**Review by acosferreira — APPROVED** [review]
lgtm

**Pull request merged**